### PR TITLE
Some fixes for ag command

### PR DESCRIPTION
**Pull request by pelijah**
* Fix r_file_path() on Windows
* Fix bug in r_core_graph_cmd()
* Use explorer.exe as picture viewer on Windows

### CODE REVIEW

File: libr/util/file.c [hunk 1]
```diff
@@ -216,6 +216,7 @@ R_API char *r_file_abspath(const char *file) {
 R_API char *r_file_path(const char *bin) {
 	char file[1024];
```
Comment (radare) [inline libr/util/file.c]: we shouldnt be using fixed-size buffers, there's a MAXFILEPATH constant for this on most systems, except HURD where you have to use the heap.. i dont see the point on using the stack for this

File: libr/util/file.c [hunk 2]
```diff
@@ -228,13 +229,19 @@ R_API char *r_file_path(const char *bin) {
...
+#endif
 	if (path_env) {
 		str = path = strdup (path_env);
 		do {
-			ptr = strchr (str, ':');
+			ptr = strchr (str, R_SYS_ENVSEP[0]);
 			if (ptr) {
 				*ptr = '\0';
-				snprintf (file, sizeof (file), "%s"R_SYS_DIR"%s", str, bin);
+				snprintf (file, sizeof (file), bin_fmt, str, bin);
```
Comment (radare) [inline libr/util/file.c]: this is also wrong. can be not null terminated path

Comment (radare) [reply libr/util/file.c]: just use the r_str_newf

File: libr/util/file.c [hunk 3]
```diff
@@ -228,13 +229,19 @@ R_API char *r_file_path(const char *bin) {
 		return NULL;
 	}
 	path_env = (char *)r_sys_getenv ("PATH");
+	bin_fmt = R_JOIN_2_PATHS ("%s", "%s");
```
Comment (radare) [inline libr/util/file.c]: using a var to hold a format string for just appending .exe hurts a bit, you can just have a const chr *extension = ""; to be set on WINDOWS and then use that variable in the format string below. you will remove like 5 lines of code and simplify the logic quite a bit

File: libr/core/cconfig.c [hunk 4]
```diff
@@ -2305,7 +2308,7 @@ R_API char* r_core_graph_cmd(RCore *core, char *r2_cmd) {
 				cmd = r_str_newf ("%s > a.dot;!dot -T%s -oa.%s a.dot;!%s a.%s", r2_cmd, ext, ext, viewer, ext);
 				free (viewer);
 			} else {
-				cmd = "?e cannot find a valid picture viewer";
+				cmd = r_str_new ("?e cannot find a valid picture viewer");
```
Comment (radare) [inline libr/core/cconfig.c]: imho this is a warning., should be printed to stderr, and return NULL

File: libr/core/cconfig.c [hunk 5]
```diff
@@ -2271,6 +2271,9 @@ static int cb_dbgsnap(void *user, void *data) {
 }
 
 static char *getViewerPath() {
+#if __WINDOWS__
+	return r_str_new ("");
```
Comment (radare) [inline libr/core/cconfig.c]: return NULL. theres no reason to return a heap allocated empty string

Comment (radare) [inline libr/core/cconfig.c]: also if you return in here, the rest of the funciton is dead code. use #else

File: libr/util/file.c (hunk 1 above)
Comment (pelijah) [reply libr/util/file.c]: It is not my code but ok I can change it too.

**Comment by radare** [comment]
yep i know. just showing my thoughs on this code. most of the time is better to rewrite wrong old code than just bugfixing with patches

**Comment by radare** [comment]
ping

**Comment by pelijah** [comment]
I am busy now. I will fix it later today or tomorrow.

**Comment by pelijah** [comment]
@radare

**Pull request merged**